packages/lodestar/src/api/rest/controllers/beacon/state/getValidators.ts

```diff
@@ -1,18 +1,33 @@
 import {ApiController} from "../../types";
-import {DefaultQuery} from "fastify";
-import {StateId} from "../../../../impl/beacon/state";
+import {IValidatorFilters, StateId, ValidatorStatus} from "../../../../impl/beacon/state";
+import {mapValidatorIndices} from "../../utils";
 
 type Params = {
   stateId: StateId;
 };
 
-export const getStateValidators: ApiController<DefaultQuery, Params> = {
+type ValidatorsQuery = {
+  indices?: string[];
+  statuses?: string[];
+};
+
+export const getStateValidators: ApiController<ValidatorsQuery, Params> = {
   url: "/states/:stateId/validators",
 
   handler: async function (req, resp) {
-    const validators = await this.api.beacon.state.getStateValidators(req.params.stateId);
+    const filters: IValidatorFilters = {};
+    if (req.query.indices) {
+      filters.indices = mapValidatorIndices(
+        this.config,
+        typeof req.query.indices === "string" ? [req.query.indices] : req.query.indices
```

Comment:
I think you should be able to tell fastify (in the fastify schema) that req.query.indices should be an array

Reply:
When i send the indices via the getStateValidators api call from the validator client as an array to the beacon node, if the array has only one item, it looks like fastify is supposed to coerce the query param to an array (as per our server setup here: https://github.com/ChainSafe/lodestar/blob/ea43a754d42831e5ad7bef2b4c3741822467926e/packages/lodestar/src/api/rest/index.ts#L54), and yet, if i send only one item in the indices array, it comes across on the beacon side as a string, even if i change the type in the schema for the getStateValidators controller.  

It looks like we also have other places where we are doing this manual array coercion (such as here: https://github.com/ChainSafe/lodestar/blob/ea43a754d42831e5ad7bef2b4c3741822467926e/packages/lodestar/src/api/rest/controllers/node/getPeers.ts#L37) so maybe this is something we should handle in a separate issue?  Or if I'm misunderstanding and/or you have another suggestion, then I'm open to that too.

Reply:
Check my other comment.